### ISSUE TITLE
UFFI Rename FFIExternalObject into FFIExternalPointer (and deprecate old name)

### ISSUE THREAD

**Issue by astares** [guillep/pharo-core#3186]
According to a Discord discussion between "hfreyther" and @estebanlm:

 - FFIExternalObject has a bad name
 - and it should be named FFIExternalPointer

![image](https://user-images.githubusercontent.com/5980033/56143131-434c7100-5fa0-11e9-8da3-5ec63094ca93.png)

So I suggested for Pharo 8 to rename **FFIExternalObject** into **FFIExternalPointer**, and create an empty subclass FFIExternalObject which we can then deprecate so we have a transition path to a better name

![image](https://user-images.githubusercontent.com/5980033/56143260-7e4ea480-5fa0-11e9-81ef-89d3539cec63.png)

**Comment by astares** [comment]
Still valid

**Comment by guillep** [comment]
This impacts the booklet https://github.com/SquareBracketAssociates/Booklet-uFFI/

:/

<blockquote><img src="https://avatars1.githubusercontent.com/u/393408?s=400&v=4" width="48" align="right"><div><img src="https://github.githubassets.com/favicons/favicon.svg" height="14"> GitHub</div><div><strong><a href="https://github.com/SquareBracketAssociates/Booklet-uFFI">SquareBracketAssociates/Booklet-uFFI</a></strong></div><div>about the new Unified FFI frameworks. Contribute to SquareBracketAssociates/Booklet-uFFI development by creating an account on GitHub.</div></blockquote>

**Comment by estebanlm** [comment]
no, it doesn't (there are no references to FFIExternalObject to modify there)
